src/c++/library/http_client.cc

```diff
@@ -930,6 +1075,145 @@ InferResultHttp::InferResultHttp(
   }
 }
 
+InferResultHttp::~InferResultHttp()
+{
+  if (binary_output_data_) {
+    return;
+  }
+
+  for (auto& buf_pair : output_name_to_buffer_map_) {
+    const uint8_t* buf{buf_pair.second.first};
+    delete buf;
+  }
+}
+
+Error
+InferResultHttp::ConvertJSONOutputToBinary(
+    triton::common::TritonJson::Value& data_json, const std::string& datatype,
+    const uint8_t** buf, size_t* buf_size) const
+{
+  const size_t element_count{data_json.ArraySize()};
+
+  if (datatype == "BOOL") {
```

Comment:
Where does this list of datatypes come from? Is it the list that triton supports?  Can/will this client be used with anything other than Triton?

Reply:
They come from here, which Triton is modeled after: https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#tensor-data-types

This client is only used for Triton.

Reply:
I'd feel a lot more comfortable if there was coded list of types (a static vector?). That is probably outside of the scope of this story, as all other cases in the code follow your same pattern.

Reply:
Yeah, we can include that in whatever giant refactor of c++ client happens.